File: Lab 2/MATLAB/Task_2.m
% Wireline Channel modelling - 0.40 mm wire

R_oc = 280;
a_c = 0.0969;
L_0 = 587.3*10^(-6);
L_inf = 426*10^(-6);
b = 1.385;
f_m = 745900;
C_inf = 50*10^(-9);
C_0 = 0*10^(-9);
c_e = 1;
g_0 = 0;
g_e = 1;

% Part 1
freq = [4, 4000, 4000000 ]*10^3;

for f = 1:3
    R1 = (R_oc^4 + a_c*(freq(f))^2)^(1/4);
    L1 = (L_0 + L_inf*(freq(f)/f_m)^b)/(1 + (freq(f)/f_m)^b);
    G1 = g_0*freq(f)^(g_e);
    C1 = C_inf + C_0*freq(f)^(- c_e);

    omega1 = 2*pi*freq(f);
    gamma1 = sqrt((R1 + 1i*L1*omega1)*(G1 + 1i*C1*omega1));
    for i = 1:1:10
        d1(i) = 10 + (i-1)*500;
        H1(i) = 10*log10(abs(exp(- gamma1*d1(i)/1000)));
    end
    
    subplot(2,2,f)
    plot(d1,H1)
    str = sprintf('Relation of H(f) with Distance at a fixed frequency of %d Hz', freq(f));
    title(str)
    xlabel('Distance(in metres)')
    ylabel('H(f)(in dB)')

end

File: Lab 2/Python/Task_2.py
import cmath
import matplotlib.pyplot as plt

# Wireless Channel modelling

G_t = 10
G_r = 1
c = 3*10**8
P_t = 1

freq1 = 9*10**8
freq2 = 24*10**8
# 900 MHz and 2.4 GHz


fig = plt.figure(figsize=(3, 3))
P_r1 = []
P_r2 = []
d = []
for i in range(0, 20):
    di = 100 + 250*i
    d.append(di)
    P_r1.append(10 * cmath.log10(P_t * 10 ** 3 * ((G_t * G_r * c ** 2) / (4 * cmath.pi * freq1 * di) ** 2)))
    P_r2.append(10 * cmath.log10(P_t * 10 ** 3 * ((G_t * G_r * c ** 2) / (4 * cmath.pi * freq2 * di) ** 2)))

plt.plot(d, P_r1, 'o-', label='900 MHz')
plt.plot(d, P_r2, 'o-', label='2.4 GHz')
plt.title('Received power(mdB) vs Distance at two frequencies')
plt.xlabel('Distance (m)')
plt.ylabel('Received power (mdB)')
plt.grid(True, which='both')
plt.axhline(y=0, color='m')
plt.legend(loc='upper right')

axes = plt.gca()
axes.set_xlim([0, None])
axes.set_ylim([None, None])

plt.show()
